eng/packages/http-client-csharp-mgmt/generator/Azure.Generator.Management/src/Utilities/CSharpTypeExtensions.cs

```diff
@@ -31,19 +31,5 @@ public static CSharpType WrapResponse(this CSharpType? type, bool isLongRunning)
                     : typeof(Response);
             }
         }
-
-        public static CSharpType UnWrap(this CSharpType type)
-        {
-            var currentType = type;
-
-            // Recursively unwrap any generic type with a single type argument
-            // This will find the innermost type regardless of nesting level or specific generic types
-            while (currentType.IsGenericType && currentType.Arguments.Count == 1)
-            {
-                currentType = currentType.Arguments[0];
-            }
-
-            return currentType;
-        }
```

Comment:
this method in some scenarios might give you unexpected result.
For instance, imagine that we have a method returning `IReadOnlyList<string>`.
Calling this method on this return type: `Task<Response<IReadOnlyList<string>>>` will give you `string`, but in fact, we should expect `IReadOnlyList<string>`.